keyboards/40percentclub/half_n_half/keymaps/Boy_314/keymap.c

```diff
@@ -36,106 +43,135 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
 	),
-	
+
 	/* RAISE
 	 * ,-------------------------------------------------------------------------------------------------.
-	 * |Reset |      |      |Up    |      |      |      |      |      |      |      |      |      |Del   |
+	 * |Reset |      |      |Up    |      |      |      |PRVWIN|CLSTAB|      |      |      |      |Del   |
 	 * |------+------+------+------+------+------+------+------+------+------+------+------+------+------|
-	 * |`     |      |Left  |Down  |Right |      |      |      |      |-     |=     |[     |]     |\     |
+	 * |`     |      |Left  |Down  |Right |      |      |NEWTAB|ALTF4 |-     |=     |[     |]     |\     |
 	 * |------+------+------+------+------+------+------+------+------+------+------+------+------+------|
 	 * |      |!     |@     |#     |$     |%     |      |      |^     |&     |*     |(     |)     |      |
 	 * |------+------+------+------+------+------+------+------+------+------+------+------+------+------|
-	 *                             |             |             |             |      
+	 *                             |             |             |             |
 	 *                             `-------------'             `-------------'
 	 */
-	[RAISE] = LAYOUT(/* Arrows, Shifted Numbers, Symbols, Delete, Reset Key */
-	  RESET,   KC_TRNS, KC_TRNS, KC_UP,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_DEL,
-	  KC_GRV,  KC_TRNS, KC_LEFT, KC_DOWN, KC_RGHT, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, KC_BSLS, 
-	  KC_TRNS, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_TRNS, KC_TRNS, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_TRNS, 
-										  KC_TRNS, 									   KC_TRNS
+	[_RAISE] = LAYOUT(/* Arrows, Shifted Numbers, Symbols, Delete, Macros */
+	  RESET,   KC_TRNS, KC_TRNS, KC_UP,   KC_TRNS, KC_TRNS, KC_TRNS, PRVWIN,  CLSTAB,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_DEL,
+	  KC_GRV,  KC_TRNS, KC_LEFT, KC_DOWN, KC_RGHT, KC_TRNS, KC_TRNS, NEWTAB,  ALTF4,   KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, KC_BSLS,
+	  KC_TRNS, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_TRNS, KC_TRNS, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_TRNS,
+										                    KC_TRNS,                            KC_TRNS
 	)
 };
 
 void tap_dance_choose_layer (qk_tap_dance_state_t *state, void *user_data) {
 	switch (state->count) {
 		case 1:
-			layer_on(LOWER);
+			layer_on(_LOWER);
 			break;
 		case 2:
-			layer_on(RAISE);
+			layer_on(_RAISE);
 			break;
 	}
 }
 
 void tap_dance_choose_layer_reset (qk_tap_dance_state_t *state, void *user_data) {
 	switch (state->count) {
 		case 1:
-			layer_off(LOWER);
+			layer_off(_LOWER);
 			break;
 		case 2:
-			layer_off(RAISE);
+			layer_off(_RAISE);
 			break;
 		case 3:
-			if (default_layer_state == DVORAK) {
-				default_layer_set(QWERTY);
-				layer_on(QWERTY);
-				layer_off(DVORAK);
+			if (default_layer_state == _DVORAK) {
+				default_layer_set(_QWERTY);
+				layer_on(_QWERTY);
```

Comment:
Actually, you shouldn't need to turn on and off the layers here.  In fact, it's more harmful than helpful. 

https://docs.qmk.fm/#/keymap

Also: https://docs.qmk.fm/#/ref_functions?id=setting-the-persistent-default-layer

Reply:
I tried the following: 
```		
case 3:
			if (default_layer_state == _DVORAK) {
				set_single_persistent_default_layer(_QWERTY);
			}
			else if (default_layer_state == _QWERTY) {
				set_single_persistent_default_layer(_DVORAK);
			}
			break;
```

But this only switches from Dvorak to Qwerty, and not back from Qwerty to Dvorak

Reply:
Well, you need to use `biton32(default_layer_state)` there, not just the `default_layer_state`, and that should detect it properly.

Reply:
Just to make sure, you've tested it, and it works how you want, correct?

Reply:
Yea, it's working as intended now.

Reply:
Awesome, glad to hear it!